include/oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h

```diff
@@ -142,6 +195,15 @@ class sycl_backend
         return global_rank_;
     }
 
+    void lazy_report(){
```

Comment:
This function is not thread safe. You might have concurrent accesses to the async_waiter array that change its size -- calls to lazy_report and submit (which calls add_waiter).